### PR TITLE
resolve environment incrementaly, including values loaded from previous files in the list

### PR DESCRIPTION
**Pull request by ndeloof**
closes https://github.com/docker/compose/issues/9980

illustration example:
```
$ cat compose.yml 
services:
  test:
    image: alpine
    command: sh -c "env"
    env_file:
      - "base.env"
      - "extra.env"
$  cat base.env 
FOO="expected"
$  cat extra.env 
FOO="${FOO:-default}"
$  docker compose run test | grep FOO
FOO=expected
```

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -356,12 +356,12 @@ type ThrottleDevice struct {
 // ShellCommand is a string or list of string args.
 //
 // When marshaled to YAML, nil command fields will be omitted if `omitempty`
-// is specified as a struct tag. Explicitly empty commands (i.e. `[]` or `''`)
+// is specified as a struct tag. Explicitly empty commands (i.e. `[]` or `”`)
```
Comment (laurazard) [inline types/types.go:359]: Is this change intentional?

File: types/types.go [hunk 2]
```diff
@@ -356,12 +356,12 @@ type ThrottleDevice struct {
...
 // will serialize to an empty array (`[]`).
 //
 // When marshaled to JSON, the `omitempty` struct must NOT be specified.
 // If the command field is nil, it will be serialized as `null`.
-// Explicitly empty commands (i.e. `[]` or `''`) will serialize to an empty
+// Explicitly empty commands (i.e. `[]` or `”`) will serialize to an empty
```
Comment (laurazard) [inline types/types.go:364]: same here

File: types/types.go (hunk 1 above)
Comment (ndeloof) [reply types/types.go]: don't know why, I get both Goland and vscode to make this change on save :'(

**Review by laurazard — APPROVED** [review]
LGTM 🥳
